### PR TITLE
Don't count unimplemented errors as extra errors in stats

### PR DESCRIPTION
**Pull request by Amareis**
**Description:**

More clearly distinction between unimplemented yet features and wrong current type checking.

### CODE REVIEW

**Comment by kdy1** [comment]
I think we should include them in the stat

**Comment by kdy1** [comment]
Ah, I saw the changes

**Comment by Amareis** [comment]
Add custom Debug impl for Stats, to not blow-up changes - unimplemented count is printed out only if there is any.

File: crates/stc_ts_type_checker/tests/tsc.rs [hunk 1]
```diff
@@ -755,7 +778,8 @@ fn do_test(file_name: &Path, spec: TestSpec, use_target: bool) -> Result<(), Std
 
     let extra_err_count = extra_errors.len();
     stats.required_error += expected_errors.len();
-    stats.extra_error += extra_err_count;
+    stats.extra_error += extra_err_count - unimplemented_count;
+    stats.unimplemented = unimplemented_count;
```
Comment (Amareis) [inline crates/stc_ts_type_checker/tests/tsc.rs:782]: I don't really know if it should be handled more structurally - with removing such errors from extra errors vector and handling it differently in debug output below. I can do it, if it's needed.

**Review by kdy1 — CHANGES_REQUESTED** [review]
I'm sorry, but I think this would interfere development

**Comment by Amareis** [comment]
Fixed

**Comment by kdy1** [comment]
I'm not going to accept this, and not sure what you mean by `fixed`

**Comment by Amareis** [comment]
Oh, just thinks you sayed about merge conflict. Okay, close it now.

**Pull request closed without merging**